app/models/user.rb

```diff
@@ -46,6 +47,12 @@ def account_id
     @account_id ||= account&.id
   end
 
+  def credential_lock
+    return @credential_lock unless @credential_lock.nil?
+
+    @credential_lock ||= user_verification&.locked
```

Comment:
can you remove the `||=` and make it `=` since it effectively does the memoization with the early return?